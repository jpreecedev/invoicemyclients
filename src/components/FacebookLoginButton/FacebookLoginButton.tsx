import React from 'react'
import clsx from 'clsx'
import { Button, Icon } from 'semantic-ui-react'

import styles from './styles.module.css'

const FacebookLoginButton = () => {
  return (
    <Button fluid size="big" className={clsx(styles.space, 'facebook')}>
      <Icon name="facebook" />
      Log In Using Facebook
    </Button>
  )
}

export { FacebookLoginButton }
