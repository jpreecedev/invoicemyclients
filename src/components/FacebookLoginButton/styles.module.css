.space {
  margin-bottom: 8px !important;
}
