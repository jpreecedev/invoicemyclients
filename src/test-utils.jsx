import React from 'react'
import { MemoryRouter as Router } from 'react-router-dom'
import { render } from '@testing-library/react'
import { createMemoryHistory } from 'history'

function renderWithRouter(
  ui,
  { route = '/', history = createMemoryHistory({ initialEntries: [route] }) } = {}
) {
  const Wrapper = ({ children }) => <Router history={history}>{children}</Router>
  return {
    ...render(ui, { wrapper: Wrapper }),
    // adding `history` to the returned utilities to allow us
    // to reference it in our tests (just try to avoid using
    // this to test implementation details).
    history
  }
}

export { renderWithRouter }
