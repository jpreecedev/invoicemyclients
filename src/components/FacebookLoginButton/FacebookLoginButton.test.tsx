import React from 'react'
import { renderWithRouter } from '../../test-utils'

import { FacebookLoginButton } from './FacebookLoginButton'

describe('<FacebookLoginButton /> tests', () => {
  it('should match snapshot', () => {
    expect(
      renderWithRouter(<FacebookLoginButton additionalClasses="test-class" />).container
    ).toMatchSnapshot()
  })
})
