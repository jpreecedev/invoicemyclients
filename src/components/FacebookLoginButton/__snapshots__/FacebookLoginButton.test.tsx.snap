// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FacebookLoginButton /> tests should match snapshot 1`] = `
<div>
  <p>
    Log in using facebook
  </p>
</div>
`;
